tools/install-monorepo.sh

```diff
@@ -15,143 +17,344 @@ function usage {
+			(echo; echo 'eval "\$(${HOMEBREW_PREFIX}/bin/brew shellenv)"') >> ${shell_rcfile}
+		fi
 
-info "Checking if pnpm is installed..."
-if ! command -v pnpm &>/dev/null; then
-	info "Installing pnpm"
-	# Don't use https://get.pnpm.io/install.sh, that doesn't play nice with different shells.
-	# And corepack will likely lose pnpm every time nvm installs a new node version.
-	curl -f https://get.pnpm.io/v6.16.js | node - add --global pnpm
-fi
-if [[ -z "$( pnpm bin --global )" ]]; then
-	info "Setting up pnpm"
-	if ! pnpm setup; then
-		warn 'pnpm has no bin dir set, and `pnpm setup` failed. Linking the Jetpack CLI may fail.'
 	else
-		# Try to read PNPM_HOME from the login shell after `pnpm setup`, as pnpm probably changed it.
-		P=$( "$SHELL" -i -c 'echo $PNPM_HOME' ) || true
-		if [[ -n "$P" ]]; then
-			export PNPM_HOME="$P"
-			if [[ ":$PATH:" != *":$PNPM_HOME:"* ]]; then
-				export PATH="$PNPM_HOME:$PATH"
+		# Run brew updates
+		if ! brew update &> /dev/null; then
+			die "Unable to update brew!"
+		fi
+	fi
+	export HOMEBREW_NO_AUTO_UPDATE=1
+	export HOMEBREW_NO_ENV_HINTS=1
+	echo "brew: available"
+}
+
+# Checks for, installs, and updates nvm.
+function do_nvm {
+	# Get latest nvm version.
+	# Based on this: https://stackoverflow.com/a/71896712
+	good_nvm_version=$(basename $(curl -fs -o/dev/null -w %{redirect_url} https://github.com/nvm-sh/nvm/releases/latest))
+
+	if [[ -f "$HOME"/.nvm/nvm.sh ]] ; then
+		# Since nvm really several shell functions, and isn't in the environment
+		# initially, we need to source it.
+		if [[ -f "$(brew --prefix nvm)/nvm.sh" ]]; then
+			# Sometimes nvm is installed via brew, which is an unsupported install
+			# route, and uses a wrapper helper script.
+			. "$(brew --prefix nvm)/nvm.sh"
+		else
+			export NVM_DIR="$HOME/.nvm" && . "$NVM_DIR"/nvm.sh --no-use
+		fi
+
+		if [[ v$(nvm --version) != "$good_nvm_version" ]]; then
+			echo "nvm: wrong version"
+			echo "Updating nvm..."
+
+			# Download latest version
+			curl -s -o- "https://raw.githubusercontent.com/nvm-sh/nvm/$good_nvm_version/install.sh" | bash - &>/dev/null &&
+
+			# Source again to get latest shell functions
+			. "$NVM_DIR/nvm.sh" --no-use
+
+			# Something's wrong, so abort.
+			if [[ "v$(nvm --version)" != "$good_nvm_version" ]]; then
+				die "Unable to update nvm!"
 			fi
+			echo "Updated nvm."
+		fi
+	else
+		echo "nvm: not found"
+		info "Installing nvm..."
+		curl -s -o- https://raw.githubusercontent.com/nvm-sh/nvm/$good_nvm_version/install.sh | bash - &>/dev/null &&
+
+		# Source to get latest shell functions
+		export NVM_DIR="$HOME"/.nvm && . "$NVM_DIR"/nvm.sh --no-use
+
+		# Something's wrong, so abort.
+		if ! has_command nvm; then
+			die "Unable to install nvm!"
 		fi
+		echo "Installed nvm."
 	fi
-fi
 
-source .github/versions.sh
-info "Installing and linking PHP $PHP_VERSION"
-brew install php@$PHP_VERSION
-brew link php@$PHP_VERSION
+	echo "nvm: available"
+}
+
+# Checks for and installs node.
```

Comment:
We only need to install sometimes. I was trying to cut down on unnecessary noise.

Reply:
Could go either way, I suppose: a little more noise in the output for a user who might be confused by it, or a bunch of complication in the code to avoid it.